src/providers/retryProvider.ts

```diff
@@ -260,12 +260,22 @@ export class RetryProvider extends ethers.providers.StaticJsonRpcProvider {
     return response;
   }
 
-  _trySend(provider: ethers.providers.StaticJsonRpcProvider, method: string, params: Array<unknown>): Promise<unknown> {
-    let promise = this._sendAndValidate(provider, method, params);
-    for (let i = 0; i < this.retries; i++) {
-      promise = promise.catch(() => delay(this.delay).then(() => this._sendAndValidate(provider, method, params)));
+  async _trySend(
+    provider: ethers.providers.StaticJsonRpcProvider,
+    method: string,
+    params: Array<unknown>
+  ): Promise<unknown> {
+    let error: unknown;
+    for (let i = 0; i < this.retries; ++i) {
```

Comment:
This is subtle, but I think with the way the loop is designed now, we're using this.retries as the number of total tries, not retries.

Put differently, in the previous loop, 2 retries would imply 3 total attempts (first with 2 retries). In the current loop, 2 retries would imply to total attempts.

I know the language here is a little confusing. Just want to make sure we're not unintentionally changing behavior.

Reply:
It was unintentional - good catch. Should also be fixed per your proposal: https://github.com/across-protocol/sdk/pull/761/commits/bbaa33ac1ba3b9d8c2d22560a3d382e2fba67308